src/app/components/elements/inputs/Password.tsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import { Input } from "reactstrap";
+
+interface PasswordProps {
+  isPasswordVisible: boolean;
+  setIsPasswordVisible: React.Dispatch<React.SetStateAction<boolean>>;
+  passwordFieldType: "New" | "Confirm" | "Current";
+  disabled?: boolean;
+  invalid?: boolean;
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onBlur?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onFocus?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  showToggleIcon?: boolean;
+  defaultValue?: string;
+  ariadescribedby?: string;
+}
+
+const Password: React.FC<PasswordProps> = (props) => {
```

Comment:
Since this is a new class, it should have some unit tests.

Reply:
Test coverage on Password.tsx is showing at 95.45% due to it being used in integration test in My Account, Registration and Login pages which I thought would be sufficient?

Reply:
Kind of, but that has some problems: the coverage is not that important, it's more important to test the different features and behaviours of your component. You may be testing some of that behaviour indirectly in other tests, but that's not clear. Also, as a rule of thumb, when you create a new component you should probably have a few tests for that component in isolation. There is a lot of logic inside this component, and since it does not have its own test, what would happen the next time someone wants to modify the behaviour of this component? How would anyone know where to add the tests for the new behaviour?

Reply:
Makes sense, I've added a bunch of unit tests for Password in isolation